vars/skipStage.groovy

```diff
@@ -404,6 +405,9 @@ boolean call(Map config = [:]) {
                    (env.BRANCH_NAME.startsWith('provider-testing') &&
                     ! startedByTimer() &&
                     ! startedByUser()) ||
+                   (env.BRANCH_NAME.startsWith('soak-testing') &&
+                    ! startedByTimer() &&
+                    ! startedByUser()) ||
```

Comment:
These need to be DRYed out with the two similar clauses above.  The branch names should be or'd and then anded and or'd by the startBy*.